src/main/java/com/synopsys/integration/alert/web/security/authentication/ldap/LdapManager.java

```diff
@@ -38,64 +42,80 @@
 import org.springframework.security.ldap.userdetails.LdapAuthoritiesPopulator;
 import org.springframework.stereotype.Component;
 
-import com.synopsys.integration.alert.common.LdapProperties;
+import com.synopsys.integration.alert.common.exception.AlertDatabaseConstraintException;
 import com.synopsys.integration.alert.common.exception.AlertLDAPConfigurationException;
+import com.synopsys.integration.alert.component.settings.SettingsDescriptor;
+import com.synopsys.integration.alert.database.api.configuration.ConfigurationAccessor;
+import com.synopsys.integration.alert.database.api.configuration.model.ConfigurationModel;
 
 @Component
 public class LdapManager {
+    private static final Logger logger = LoggerFactory.getLogger(LdapManager.class);
     private LdapContextSource contextSource;
     private LdapAuthenticationProvider authenticationProvider;
-    private LdapProperties currentConfiguration;
+    private final ConfigurationAccessor configurationAccessor;
 
     @Autowired
-    public LdapManager(final LdapProperties ldapProperties) {
-        this.currentConfiguration = ldapProperties;
+    public LdapManager(final ConfigurationAccessor configurationAccessor) {
+        this.configurationAccessor = configurationAccessor;
     }
 
     public boolean isLdapEnabled() {
-        return currentConfiguration != null && currentConfiguration.isEnabled();
-    }
+        boolean enabled = false;
+        try {
+            enabled = Boolean.valueOf(getFieldValueOrEmpty(getCurrentConfiguration(), SettingsDescriptor.KEY_LDAP_ENABLED));
+        } catch (final AlertDatabaseConstraintException | AlertLDAPConfigurationException ex) {
+            logger.error("Error checking if Ldap is enabled", ex);
+        }
 
-    public LdapProperties getCurrentConfiguration() {
-        return currentConfiguration;
+        return enabled;
     }
 
-    public LdapAuthenticationProvider getAuthenticationProvider() throws AlertLDAPConfigurationException {
-        if (authenticationProvider == null) {
-            updateConfiguration(currentConfiguration);
+    public ConfigurationModel getCurrentConfiguration() throws AlertDatabaseConstraintException, AlertLDAPConfigurationException {
+        final List<ConfigurationModel> configurations = configurationAccessor.getConfigurationsByDescriptorName(SettingsDescriptor.SETTINGS_COMPONENT);
```

Comment:
This could become: 
```
return configurationAccessor.getConfigurationsByDescriptorName(SettingsDescriptor.SETTINGS_COMPONENT)
.stream()
.findFirst()
.orElseThrow(() -> new AlertLDAPConfigurationException("more than 1 settings configuration present"));
```